pkg/runtime/serializer/recognizer/recognizer.go

```diff
@@ -27,18 +27,38 @@ import (
 
 type RecognizingDecoder interface {
 	runtime.Decoder
+	// AccurateRecognizer should be true if this decoder can unambigously determine
+	// whether the content is known to it. Allows RecognizingDecoders to adjust certainty.
+	AccurateRecognizer() bool
+	// RecognizesData should return true if the input provided in the provided reader
+	// belongs to this decoder, or an error if the data could not be read or is ambiguous.
 	RecognizesData(peek io.Reader) (bool, error)
```

Comment:
Maybe just consider changing this interface to something like:
* definitely yes I handle this data and no one else could
* I might be able to handle this, but I'm not confident
* I definitely can't handle this data.